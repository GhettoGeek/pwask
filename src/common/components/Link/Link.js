import React from 'react'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'
import { Link } from '@material-ui/core'

function EnhancedLink({ to, children, ...props }) {
  return (
    <Link component={RouterLink} to={to} {...props}>
      {children}
    </Link>
  )
}

EnhancedLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape(),
  ]),
}

EnhancedLink.defaultProps = {
  children: '',
}

export default EnhancedLink
