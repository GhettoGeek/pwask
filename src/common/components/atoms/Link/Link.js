import React from 'react'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'
import { Link } from '@material-ui/core'

function EnhancedLink({ to, children }) {
  return (
    <Link component={RouterLink} to={to}>
      {children}
    </Link>
  )
}

EnhancedLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape(),
  ]),
}

EnhancedLink.defaultProps = {
  children: '',
}

export default EnhancedLink
