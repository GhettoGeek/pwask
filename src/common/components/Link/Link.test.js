import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import Link from './Link'

describe('Link', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Link to="/">link</Link>
      </MemoryRouter>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
