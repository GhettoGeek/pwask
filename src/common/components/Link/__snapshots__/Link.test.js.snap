// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly the component 1`] = `"<a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"/\\">link</a>"`;
